### ISSUE TITLE
Leaking / "bleeding" masks

### ISSUE THREAD

**Issue by brr53** [ultralytics/ultralytics#3274]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I've noticed the yolov8/v5 masks often leaks out into the background of the class/object.

At first, I thought it could be due to the training examples. But, I've trained a model with this specific image annotated and the trained model still cant get the mask correct (it covers part of the road still). I also tried with other images that have the same phenomenon. I even tried training a model using **only** the images that leak - No success.

I only can get the model to not detect the road as a part of the motorcycle when using "yolov8x-seg" as the base when training. but that model is **far** too overkill for an easy task such as this image. 

Is yolov8m, s, and n missing a specific necessary layer for this that yolov8x has? Or what could be happening here? 
![image](https://github.com/ultralytics/ultralytics/assets/60824624/deda99ee-c8b8-491a-b8d9-c65c830d5c88)
![image](https://github.com/ultralytics/ultralytics/assets/60824624/e5eed423-78f9-4d00-8f58-90bd1d5f7421)

### Additional

I've tried many things without success such as using Adam vs SGD, etc. I'll try any suggestions you have and report feedback.

I'm training and inferring using 480, 480 img_sz

Thank you.

**Comment by glenn-jocher** [comment]
@brr53 you likely need more training data. See https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/ for details and best practices for instances and images counts.

**Comment by brr53** [comment]
@glenn-jocher Thank you for the suggestion.

I've tried training the model using 10000 labeled datas with the same style of motorcycles and backgrounds. I've also tried the default yolo models (coco) with "motorcycle". Same result - only yolov8x-seg.pt works smoothly.

The main concern is, I've trained the model under default training settings with the same image I've used in the detection meaning it already has the correct answer. But the mask still bleeds. Any additional suggestions you may have? Or any reasons this may occur?
